engine/runtime/src/main/scala/org/enso/interpreter/instrument/NotificationHandler.scala

```diff
@@ -131,6 +145,12 @@ object NotificationHandler {
       )
     }
 
+    /** @inheritdoc */
+    override def serializeModule(module: QualifiedName): Unit =
+      endpoint.sendToSelf(
+        Api.Request(Api.SerializeModule(module))
```

Comment:
Why do we need to modify `NotificationHandler` at all? Why does it send a message to self? What is this good for?

Reply:
The notification handler is a way to communicate with the [execution service](https://github.com/enso-org/enso/blob/126a4668720141e235aaa54c6dad41a10a82ac60/engine/runtime/src/main/scala/org/enso/interpreter/instrument/Handler.scala#L95-L110) that runs the background processing queue. Execution service is initialized only in interactive mode; otherwise, the compiler does not know if it exists.